package-config.cmake

```diff
@@ -70,21 +70,49 @@ MACRO(ADD_PROJECT_DEPENDENCY)
 
   # add dependency to the generated .pc
   # ref https://github.com/jrl-umi3218/jrl-cmakemodules/pull/335
-  cmake_parse_arguments(PARSED_ARGN "" "PKG_CONFIG_REQUIRES" "" ${ARGN})
+  # Add options OPTIONAL or REQUIRED
+  cmake_parse_arguments(PARSED_ARGN
+    "OPTIONAL;REQUIRED"
+    "PKG_CONFIG_REQUIRES"
+    ""
+    ${ARGN})
+
   IF(PARSED_ARGN_PKG_CONFIG_REQUIRES)
-    _ADD_TO_LIST_IF_NOT_PRESENT(_PKG_CONFIG_REQUIRES "${PARSED_ARGN_PKG_CONFIG_REQUIRES}")
-    _ADD_TO_LIST_IF_NOT_PRESENT(_PKG_CONFIG_DEP_NOT_FOR_CONFIG_CMAKE "${PARSED_ARGN_PKG_CONFIG_REQUIRES}")
+    _ADD_TO_LIST_IF_NOT_PRESENT(_PKG_CONFIG_REQUIRES
+      "${PARSED_ARGN_PKG_CONFIG_REQUIRES}")
+    _ADD_TO_LIST_IF_NOT_PRESENT(_PKG_CONFIG_DEP_NOT_FOR_CONFIG_CMAKE
+      "${PARSED_ARGN_PKG_CONFIG_REQUIRES}")
   ENDIF()
 
-  string(REPLACE ";" " " PACKAGE_ARGS "${PARSED_ARGN_UNPARSED_ARGUMENTS}")
-  list(APPEND _PACKAGE_CONFIG_DEPENDENCIES_FIND_PACKAGE "  find_package(${PACKAGE_ARGS})")
-  list(APPEND _PACKAGE_CONFIG_DEPENDENCIES_FIND_DEPENDENCY "  find_dependency(${PACKAGE_ARGS})")
-  find_package(${PARSED_ARGN_UNPARSED_ARGUMENTS})
+  string(REPLACE ";" " " PACKAGE_ARGS
+    "${PARSED_ARGN_UNPARSED_ARGUMENTS}")
+
+  # If REQUIRED is not specified consider the package as optional
+  # search for the package in a quiet mode.
+  IF(PARSED_ARGN_OPTIONAL OR NOT PARSED_ARGN_REQUIRED)
+    find_package(${PARSED_ARGN_UNPARSED_ARGUMENTS} QUIET)
+  # Else generate an error
+  ELSE(PARSED_ARGN_OPTIONAL OR NOT PARSED_ARGN_REQUIRED)
+    find_package(${PARSED_ARGN_UNPARSED_ARGUMENTS} REQUIRED)
+  ENDIF(PARSED_ARGN_OPTIONAL OR NOT PARSED_ARGN_REQUIRED)
+
+  # If the package has been found provides the dependency
+  IF(${${ARGV0}_FOUND})
+
+    list(APPEND _PACKAGE_CONFIG_DEPENDENCIES_FIND_PACKAGE
+      "  find_package(${PACKAGE_ARGS})")
+    list(APPEND _PACKAGE_CONFIG_DEPENDENCIES_FIND_DEPENDENCY
+      "  find_dependency(${PACKAGE_ARGS})")
```

Comment:
The generated file will miss the `REQUIRED` argument since you consume it in `cmake_parse_arguments`

Reply:
Yes we would need to test PARSED_ARGN_REQUIRED to add or not the REQUIRED parameter.